### PR TITLE
Remove deprecated ServiceInstanceSupplier

### PR DESCRIPTION
**Pull request by fitzoh**
While familiarizing myself with the codebase, I noticed that `ServiceInstanceSupplier` was deprecated in August 2019 (#595), and also noticed that the current version is 3.0.0-SNAPSHOT.

Sounds like a good time for a deprecation cleanup to me.

### CODE REVIEW

File: spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/LoadBalancerTests.java [hunk 1]
```diff
@@ -123,11 +122,11 @@ public void emptyHosts() {
 	@Test
 	public void staticConfigurationWorks() {
```
Comment (m1ngyuan) [inline spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/LoadBalancerTests.java:123]: Duplicate test case staticconfigurationworkswithserviceinstancelistsupplier

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java [hunk 2]
```diff
@@ -76,23 +70,6 @@ public ServiceInstanceListSupplier discoveryClientServiceInstanceListSupplier(
 					.withCaching().build(context);
 		}
 
-		@Bean
```
Comment (m1ngyuan) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:79]: I think it's just to replace the deprecated class, not to delete it completely

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java [hunk 3]
```diff
@@ -109,23 +86,6 @@ public ServiceInstanceListSupplier discoveryClientServiceInstanceListSupplier(
...
```
Comment (m1ngyuan) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: I think it's just to replace the deprecated class, not to delete it completely

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java (hunk 2 above)
Comment (m1ngyuan) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:79]: I think it's just to replace the deprecated class, not to delete it completely

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java (hunk 3 above)
Comment (m1ngyuan) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: I think it's just to replace the deprecated class, not to delete it completely

Comment (spencergibb) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: The class is already deprecated in 2.2.x and can be removed in master.

Comment (m1ngyuan) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: But we still need serviceinstancelistsupplier bean. We can modify it in this way:
@Bean
		@ConditionalOnBean(ReactiveDiscoveryClient.class)
		@ConditionalOnMissingBean
		public ServiceInstanceListSupplier discoveryClientServiceInstanceSupplier(
				ReactiveDiscoveryClient discoveryClient, Environment env,
				ApplicationContext context) {
			DiscoveryClientServiceInstanceListSupplier delegate = new DiscoveryClientServiceInstanceListSupplier(
					discoveryClient, env);
			ObjectProvider<LoadBalancerCacheManager> cacheManagerProvider = context
					.getBeanProvider(LoadBalancerCacheManager.class);
			if (cacheManagerProvider.getIfAvailable() != null) {
				return new CachingServiceInstanceListSupplier(delegate,
						cacheManagerProvider.getIfAvailable());
			}
			return delegate;
		}

Comment (spencergibb) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: That is already in this class

Comment (spencergibb) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java:112]: no

**Pull request closed without merging**